### PR TITLE
Add RemoveRequiredStatusChecks API and tests

### LINKED ISSUE #1770: Need a function to remove status check contexts
I want to remove all the status checks that are added under my branch protection, to do so I tried using this `UPDATE` API call with the array of status checks as empty: https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#update-status-check-protection
but I'm not able to achieve what I wanted.
Then I tried using this recent `DELETE` API to remove status checks, 
https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#remove-status-check-protection
It worked but there is no equivalent function in this go package. Can you add the same?

### PR DESCRIPTION
**Pull request by n1lesh**
Fixes #1770

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Very nice, @n1lesh ... thank you!
LGTM.

Awaiting second LGTM before merging.

**Review by wesleimp — APPROVED** [review]
LGTM 👌🏼

**Comment by gmlewis** [comment]
Thank you, @wesleimp !
Merging.

**Pull request merged**
